tools/base/libboard/board_rc/board_rc.h

```diff
@@ -0,0 +1,121 @@
+// Copyright(c) 2019, Intel Corporation
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef __FPGA_BOARD_RC_H__
+#define __FPGA_BOARD_RC_H__
+
+#include <opae/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+
+typedef struct _powerdown_cause {
+	uint8_t _header[3]; // Ignored
+	uint8_t completion_code;
+	uint8_t iana[3];
+	uint8_t count;
+	uint8_t message[40];
+} powerdown_cause;
+
+typedef struct _reset_cause {
+	uint8_t _header[3];// Ignored
+	uint8_t completion_code;
+	uint8_t iana[3];
+	uint8_t reset_cause;
+} reset_cause;
+
+
+typedef enum {
+	CHIP_RESET_CAUSE_POR = 0x01,
+	CHIP_RESET_CAUSE_EXTRST = 0x02,
+	CHIP_RESET_CAUSE_BOD_IO = 0x04,
+	CHIP_RESET_CAUSE_WDT = 0x08,
+	CHIP_RESET_CAUSE_OCD = 0x10,
+	CHIP_RESET_CAUSE_SOFT = 0x20,
+	CHIP_RESET_CAUSE_SPIKE = 0x40,
+} ResetCauses;
+
+typedef struct _device_id {
```

Comment:
Consider a better name for this.  device_id collides with the notion of PCIe device ID.  Also, there is a struct member named device_id on line 66.  This may cause confusion.

Reply:
fixed